src/test/java/bio/terra/app/controller/JobsApiControllerTest.java

```diff
@@ -15,29 +15,35 @@
 import bio.terra.app.usermetrics.BardClient;
 import bio.terra.common.SqlSortDirection;
 import bio.terra.common.category.Unit;
+import bio.terra.common.fixtures.AuthenticationFixtures;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
 import bio.terra.model.JobModel;
 import bio.terra.model.JobModel.JobStatusEnum;
+import bio.terra.service.auth.iam.PolicyMemberValidator;
+import bio.terra.service.dataset.AssetModelValidator;
+import bio.terra.service.dataset.DatasetRequestValidator;
+import bio.terra.service.dataset.IngestRequestValidator;
 import bio.terra.service.job.JobService;
 import bio.terra.service.job.JobService.JobResultWithStatus;
+import bio.terra.service.snapshot.SnapshotRequestValidator;
 import java.util.List;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.HttpStatus;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
-@AutoConfigureMockMvc
+@ContextConfiguration(classes = {JobsApiController.class, GlobalExceptionHandler.class})
+@WebMvcTest
 @ActiveProfiles({"google", "unittest"})
-@Category(Unit.class)
-public class JobsApiControllerTest {
+@Tag(Unit.TAG)
+class JobsApiControllerTest {
```

Comment:
Was this one of the hard cases to migrate?

And, I'll shout out for others that we were recently talking about how `@WebMvcTest`s don't give you the strict stubbing protections we'd expect: https://broadinstitute.slack.com/archives/C03B4RKTA5B/p1713971590899839?thread_ts=1713969215.339079&cid=C03B4RKTA5B

Reply:
This wasn't *too* bad, `BigQueryPdaoUnitTest` was probably the most work.

In this case the original test was constructing a complete application context, so it's using `live` versions of the validation classes. I replaced the live classes with mocks, and the tests still pass, so there were no tests that relied on real validation objects. Alternatively, if `JobsApiController` does rely on validators, we could instead add the ones it uses to the `@ContextConfiguration` for this test.